pkg/services/publicdashboards/api/api.go

```diff
@@ -149,6 +149,11 @@ func (api *Api) QueryPublicDashboard(c *models.ReqContext) response.Response {
 		return response.Error(http.StatusBadRequest, "invalid panel ID", err)
 	}
 
+	reqDTO := &PublicDashboardQueryDTO{}
```

Comment:
Is this enough validation of user input? Do we need to sanity check these values? What happens if we get 0, -1, or 999999999 ?

Reply:
Also- for the sake of discussion, our original security model relies on not taking user input.

How complicated is it to calculate this on the backend?

Reply:
Added validation for negative values in the request and calculate safe values for interval.